### ISSUE TITLE
listeners for image loading 

### ISSUE THREAD

**Issue by vipulasri** [renefloor/flutter_cached_network_image#30]
Is there a way to have listeners for image loading: success as well as error listener?

**Comment by renefloor** [comment]
If you use the CachedNetworkImageProvider directly you can listen to success and errors, the CachedNetworkImage makes use of that.

**Comment by vipulasri** [comment]
But I want to use `CachedNetworkImage` so that I can have the capability of placeholder, error, fadeIn etc. 
I want this change, to have success and failure listener with `CachedNetworkImage`

**Comment by GeoTob** [comment]
I suggest to provide a public `getCodec` method like below and use it in `CachedNetworkImage.load` (instead of the private `_loadAsync`). Alternatively one could simply make `_loadAsync` a public method so users can chain a custom success callback.

```
Future<ui.Codec> getCodec(CachedNetworkImageProvider key) async => _loadAsync(key);
```

Then users could simply override the `load` and `getCodec` methods with any sort of placeholder or fadeIn logic they like. Override `load` to display placeholder, override `getCoded.then` to remove placeholders.

**Comment by renefloor** [comment]
Implemented with builders

**Comment by iqbaltld** [comment]
> If you use the CachedNetworkImageProvider directly you can listen to success and errors, the CachedNetworkImage makes use of that.

but how to know when image is loaded?
I want to show a close button only if the image was loaded.

**Comment by VITAL1E** [comment]
> > If you use the CachedNetworkImageProvider directly you can listen to success and errors, the CachedNetworkImage makes use of that.
> 
> but how to know when image is loaded?
> I want to show a close button only if the image was loaded.

Hey! Did you find out how to know when image is loaded? I am facing a similar issue right now.

**Comment by Ultranmus** [comment]
I'm also looking for a solution to retrieve ImageInfo, specifically for obtaining the image's height and width. Once this information is available, I want to update the view accordingly. Is there no solution for this currently?

I tried flutter_cache_manager to directly get file and retrieve image size from it but it did not work if image failed to load initially.

I attempted to set a listener for a File using the .getFile() method in flutter_cache_manager. However, when the image fails to load initially and is later loaded and cached upon navigating to the second screen, the .getFile() listener doesn't trigger when went back to previous screen. Consequently, I'm unable to calculate the image size in such scenarios.
